[Bug]: torii-client buid not working on nextjs/vercel

### Dojo.js Version

0.7.0-alpha.1

### What happened?

Building in Vercel when using next.js fails because torii-client built products breaks node.js [Mandatory file extensions](https://nodejs.org/api/esm.html#mandatory-file-extensions) rules.

> A file extension must be provided when using the import keyword to resolve relative or absolute specifiers. Directory indexes (e.g. './startup/index.js') must also be fully specified.
> This behavior matches how import behaves in browser environments, assuming a typically configured server.


### What is the expected behavior?

All packages are built with `tsup --dts-resolve`, only torii-client is built with `tsc`, and is the only one causing problems. 

There are ways to override this error locally, but Vercel seems to be more *sensible*, and deployment always fails.

### Steps to reproduce

* build torii-client
* check contents of `dist/index.js`

```
$ cat dist/index.js
export * from "@dojoengine/torii-wasm";
export * from "./utils";
```

* the second line, importing `./utils`, breaks node.js the Mandatory file extensions rules.


### Trace

My Vercel build log:

```
[20:10:09.625] > pistols@0.1.0 build /vercel/path0/client
[20:10:09.626] > npx tsc -p tsconfig.json && NODE_OPTIONS='--experimental-wasm-modules --experimental-specifier-resolution=node' next build
[20:10:09.626] 
[20:10:17.198]    Linting and checking validity of types ...
[20:10:25.714]    Creating an optimized production build ...
[20:10:46.723]  ✓ Compiled successfully
[20:10:46.724]    Collecting page data ...
[20:10:48.622] (node:357) ExperimentalWarning: Importing WebAssembly modules is an experimental feature and might change at any time
[20:10:48.623] (Use `node --trace-warnings ...` to show where the warning was created)
[20:10:48.625] (node:358) ExperimentalWarning: Importing WebAssembly modules is an experimental feature and might change at any time
[20:10:48.625] (Use `node --trace-warnings ...` to show where the warning was created)
[20:10:48.681] (node:351) ExperimentalWarning: Importing WebAssembly modules is an experimental feature and might change at any time
[20:10:48.681] (Use `node --trace-warnings ...` to show where the warning was created)
[20:10:49.560] Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/vercel/path0/client/node_modules/.pnpm/@dojoengine+torii-client@0.7.0-alpha.1/node_modules/@dojoengine/torii-client/dist/utils' imported from /vercel/path0/client/node_modules/.pnpm/@dojoengine+torii-client@0.7.0-alpha.1/node_modules/@dojoengine/torii-client/dist/index.js
[20:10:49.560]     at finalizeResolution (node:internal/modules/esm/resolve:265:11)
[20:10:49.560]     at moduleResolve (node:internal/modules/esm/resolve:933:10)
[20:10:49.560]     at defaultResolve (node:internal/modules/esm/resolve:1157:11)
[20:10:49.561]     at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:390:12)
[20:10:49.561]     at ModuleLoader.resolve (node:internal/modules/esm/loader:359:25)
[20:10:49.561]     at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:234:38)
[20:10:49.561]     at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:87:39)
[20:10:49.561]     at link (node:internal/modules/esm/module_job:86:36) {
[20:10:49.561]   code: 'ERR_MODULE_NOT_FOUND',
[20:10:49.561]   url: 'file:///vercel/path0/client/node_modules/.pnpm/@dojoengine+torii-client@0.7.0-alpha.1/node_modules/@dojoengine/torii-client/dist/utils'
[20:10:49.561] }
[20:10:49.566] 
[20:10:49.567] > Build error occurred
[20:10:49.567] Error: Failed to collect page data for /internal/snapshot
[20:10:49.567]     at /vercel/path0/client/node_modules/.pnpm/next@13.5.6_react-dom@18.3.1_react@18.3.1__react@18.3.1_sass@1.77.4/node_modules/next/dist/build/utils.js:1171:15 {
[20:10:49.567]   type: 'Error'
[20:10:49.567] }
[20:10:49.666]  ELIFECYCLE  Command failed with exit code 1.
[20:10:49.688] Error: Command "pnpm run build" exited with 1
[20:10:50.173] 
```

### Is there an existing issue for this?

- [X] I have searched the existing issues and verified no issue exits for this problem.

--- a/packages/torii-client/package.json
+++ b/packages/torii-client/package.json
@@ -8,7 +8,7 @@
     "main": "dist/index.js",
     "type": "module",
     "scripts": {
-        "build": "tsc"
+        "build": "tsup --dts-resolve"
     },
     "exports": {
         ".": {

